src/ripple/peerfinder/impl/Livecache.h

```diff
@@ -482,7 +483,9 @@ Livecache <Allocator>::hops_t::shuffle()
         v.reserve (list.size());
         std::copy (list.begin(), list.end(),
             std::back_inserter (v));
-        std::random_shuffle (v.begin(), v.end());
+        std::random_device rng;
+        std::mt19937 urng{rng()};
+        std::shuffle (v.begin(), v.end(), urng);
```

Comment:
This effectively uses `random_device` as the generator.  Instead a generator should be seeded once and used repeatedly.  Fortunately this is already a solved problem in rippled.   We should just use `default_prng()` from `<ripple/basics/random.h>`.

Reply:
@HowardHinnant very cool that this is already a solved problem inside rippled. Read through the code in `random.h` and it leaves me wondering whether we reach out to a 3rd party for a cryptographically secure rng (I suspect so).

In any regard, thanks for pointing it out. I've just pushed up a commit that uses `default_prng`.

Reply:
@JoeLoser we have `crypto_prng` which calls down to OpenSSL to retrieve cryptographically secure randomness (although, one of these days, I want to remove that dependency).

See https://github.com/ripple/rippled/blob/develop/src/ripple/crypto/csprng.h and https://github.com/ripple/rippled/blob/develop/src/ripple/crypto/impl/csprng.cpp for the details.

Actually, this got me thinking. A bit of extra work could allow us to prevent accidentally passing a non-CSPRNG engine into functions that **MUST** use only CSPRNG engines. One option is trait-based, but other implementations are possible:

We drop the the following `random.h`:
```template <typename Engine> class is_csprng : public std::false_type {};```

We then drop the following `csprng.h`:
```template <> class is_csprng<csprng_engine> : public std::true_type {};```

I wonder if this is worth doing.

Reply:
@nbougalis thanks for the pointers. I'm not surprised at the OpenSSL dependency.  I quickly glanced at `csprng_engine`, but I'll look more closely at it this weekend as the devil is in the details of mixing entropy.

I'm all for type safety to prevent errors. I do think your traits approach should be sufficient from the little I know about our existing usage.  As a group, we would also probably want to agree on a list of which functions we deem require CSPRNG engines (are they all in the `crypto` source folder?) before we employ `static_assert`ing on `is_csprng`.

Reply:
Happy to have another review of that code.